Add a full-stop to all admin UI help texts where it is missing

ahus1 opened the pull request:
Closes #36211

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

ahus1 committed:
Add a full-stop to all admin UI help texts where it is missing

Closes #36211

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Co-authored-by: andymunro <48995441+andymunro@users.noreply.github.com>
Signed-off-by: Alexander Schwartz <alexander.schwartz@gmx.net>

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -638,9 +638,9 @@ createUser=Create user
-signDocumentsHelp=Should SAML documents be signed by the realm?
+useKerberosForPasswordAuthenticationHelp=Use Kerberos login module for authenticating username/password against Kerberos server instead of authenticating against LDAP server with Directory Service API.
+guiOrderHelp=Specify order of the provider in GUI (such as in Consent page) as integer.
+signDocumentsHelp=Should SAML documents be signed by the realm?.
```
You don't end a question with a period.  This one should be unchanged.

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -825,18 +825,18 @@ membershipUserLdapAttributeHelp=Used just if Membership Attribute Type is UID. I
 clientScopesRolesScope=If there is no role scope mapping defined, each user is permitted to use this client scope. If there are role scope mappings defined, the user must be a member of at least one of the roles.
 passwordPoliciesHelp.notUsername=The password cannot match the username.
 passwordPoliciesHelp.notContainsUsername=The password cannot contains the username.
 removeConfirm_one=Are you sure you want to remove this group
```
```suggestion
removeConfirm_one=Are you sure you want to remove this group?
```

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -825,18 +825,18 @@ membershipUserLdapAttributeHelp=Used just if Membership Attribute Type is UID. I
 removeConfirm_one=Are you sure you want to remove this group
 createUserProviderSuccess=User federation provider successfully created
-countHelp=Specifies how many clients can be created using the token
+countHelp=Specifies how many clients can be created using the token.
```
This is not a complete sentence.  I'm not sure if it's proper to end with a period.

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -955,7 +955,7 @@ placeholderText=Select one
 authDefaultActionTooltip=If enabled, any new user will have this required action assigned to it.
 validateBindCredentials=You must enter the password of the LDAP admin
-evictionMinuteHelp=Minute of the hour the entry will become invalid
+evictionMinuteHelp=Minute of the hour the entry will become invalid.
```
This is not a complete sentence.  I'm not sure if it should have a period.

andymunro replied:
```suggestion
evictionMinuteHelp=Minute of the hour the entry will become invalid
```

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -994,12 +994,12 @@ orderChangeErrorUserFed=Could not change the priority order of user federation p
 deleteClientPolicyConfirmTitle=Delete policy?
 validateRdnLdapAttribute=You must enter an RDN LDAP attribute
-policyUrlHelp=URL that the Relying Party Client provides to the End-User to read about the how the profile data will be used
+policyUrlHelp=URL that the Relying Party Client provides to the End-User to read about the how the profile data will be used.
```
This is not a complete sentence.  I'm not sure if it should have a period.

andymunro replied:
```suggestion
policyUrlHelp=URL that the Relying Party Client provides to the End-User to read about the how the profile data will be used
```

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -994,12 +994,12 @@ orderChangeErrorUserFed=Could not change the priority order of user federation p
 clientRoles=Client roles
 removeRoles=Remove roles
-flowNameDescriptionHelp=Help text for the name description of the new flow
+flowNameDescriptionHelp=Help text for the name description of the new flow.
```
This is not a complete sentence.  I'm not sure if it should have a period.

andymunro replied:
```suggestion
flowNameDescriptionHelp=Help text for the name description of the new flow
```

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1020,7 +1020,7 @@ frontchannelLogoutSessionRequired=Front-channel logout session required
 testConnectionHint.withoutEmailAction=Configure e-mail address
 webAuthnUpdateError=Could not update webauthn policies due to {{error}}
-paginationHelp=Whether the LDAP server supports pagination
+paginationHelp=Whether the LDAP server supports pagination.
```
This is not a complete sentence.  I'm not sure if it should have a period.

andymunro replied:
```suggestion
paginationHelp=Whether the LDAP server supports pagination
```

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -938,7 +938,7 @@ unlinkUsers=Unlink users
 emailVerification=Email Verification
 configSaveError=Could not save the config\: {{error}}
-clientAuthenticatorTypeHelp=Client Authenticator used for authentication of this client against Keycloak server
+clientAuthenticatorTypeHelp=Client Authenticator used for authentication of this client against Keycloak server.
```
This is not a complete sentence.  I'm not sure if it should have a period.

andymunro replied:
```suggestion
clientAuthenticatorTypeHelp=Client Authenticator used for authentication of this client against Keycloak server
```

ssilvert commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -884,7 +884,7 @@ deleteConfirmTitle_one=Delete group?
 webAuthnUpdateSuccess=Updated webauthn policies successfully
 authorizationSignedResponseAlg=Authorization response signature algorithm
-mapperTypeFullNameLdapMapperHelp=Used to map the full-name of a user from single attribute in LDAP (usually 'cn' attribute) to firstName and lastName attributes of UserModel in Keycloak DB
+mapperTypeFullNameLdapMapperHelp=Used to map the full-name of a user from single attribute in LDAP (usually 'cn' attribute) to firstName and lastName attributes of UserModel in Keycloak DB.
```
This is not a complete sentence.  I'm not sure if it should have a period.

andymunro replied:
I agree with Stan.

andymunro replied:
```suggestion
mapperTypeFullNameLdapMapperHelp=Used to map the full-name of a user from single attribute in LDAP (usually 'cn' attribute) to firstName and lastName attributes of UserModel in the Keycloak database
```

ssilvert reviewed (commented):
I started to review this, but I'm not sure if it's worth the effort to go through all the messages.

There are some messages that are complete sentences and therefore should definitely end in a period.  

However, it's common for help text to simply describe a field without using a complete sentence.  I'm not sure if this is considered good practice or not and I'm not sure if those sentence fragments should end in a period.

If we really want to go the extra mile on this one then we should consult Andy.

ahus1 commented:
This originated from the translations which where were adding to several help messages a full stop at the end, and then our translation tool raised its concerns that the original translation didn't contain a full stop. Still I agree that incomplete sentences might need to be handled differently. 

I fixed the obvious mistakes in another commit. 

@andymunro - can you please comment here if there is a best practice to handle those incomplete sentences in help texts? 

See below for an example where I added a full stop at the end, and @ssilvert correctly states that this is not a full sentence. 

![image](https://github.com/user-attachments/assets/b84b0bcd-bfdd-4e50-aac8-96ee74581a51)

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -424,7 +424,7 @@ logoutServiceRedirectBindingURL=Logout Service Redirect Binding URL
 emptyMappersInstructions=If you want to add mappers, please click the button below to add some predefined mappers or to configure a new mapper.
 dayMonth=Day
-clientRolesHelp=The condition checks whether one of the specified client roles exists on the client to determine whether the policy is applied. This effectively allows client administrator to create client role of specified name on the client to make sure that particular client policy will be applied on requests of this client. Condition is checked during most of OpenID Connect requests (Authoriz ...
+clientRolesHelp=The condition checks whether one of the specified client roles exists on the client to determine whether the policy is applied. This effectively allows client administrator to create client role of specified name on the client to make sure that particular client policy will be applied on requests of this client. Condition is checked during most of OpenID Connect requests (Authoriz ...
```
```suggestion
clientRolesHelp=The condition checks whether one of the specified client roles exists on the client to determine whether the policy is applied. This condition effectively allows the client administrator to create a client role of a specified name on the client to make sure that particular client policy will be applied on requests of this client. The condition is checked during most OpenID Connect requests (Authorization requests, token requests, introspection endpoint request, and so on).
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -651,10 +651,10 @@ deleteResource=Permanently delete resource?
 clientSecret=Client Secret
 inputType=Input type
-claimHelp=Name of claim to search for in token. You can reference nested claims by using a '.', i.e. 'address.locality'. To use dot (.) literally, escape it with backslash. (\\.)
+claimHelp=Name of claim to search for in token. You can reference nested claims by using a '.', i.e. 'address.locality'. To use dot (.) literally, escape it with backslash (\\.).
```
```suggestion
claimHelp=Name of claim to search for in token. You can reference nested claims by using a '.', for example, 'address.locality'. To use dot (.) literally, escape it with backslash (\\.).
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1043,7 +1043,7 @@ attributeNameHelp=Name of attribute to search for in assertion. You can leave th
 linkAccountTitle=Link account to {{provider}}
 invalidateRotatedSuccess=Rotated secret successfully removed
-userSessionAttributeHelp=Name of user session attribute you want to hardcode
+userSessionAttributeHelp=Name of user session attribute you want to hardcode.
```
```suggestion
userSessionAttributeHelp=Name of user session attribute you want to hardcode
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1061,7 +1061,7 @@ host=Host
 backchannelLogoutRevokeOfflineSessions=Backchannel logout revoke offline sessions
 supportedApplications=Supported applications
-shortVerificationUriTooltipHelp=If set, this value will be return as verification_uri in Device Authorization flow. This uri need to redirect to {server-root}/realms/{realm}/device
+shortVerificationUriTooltipHelp=If set, this value will be return as verification_uri in Device Authorization flow. This uri need to redirect to {server-root}/realms/{realm}/device.
```
```suggestion
shortVerificationUriTooltipHelp=If set, this value will be returned as verification_uri in the Device Authorization flow. This uri needs to redirect to {server-root}/realms/{realm}/device.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1095,13 +1095,13 @@ clientSaveError=Client could not be updated\: {{error}}
 certificate=Certificate
 deleteConfirmExecutionMessage=Are you sure you want to permanently delete the execution "<1>{{name}}</1>".
-offlineSessionMaxLimitedHelp=Enable offline session max
+offlineSessionMaxLimitedHelp=Enable offline session maximum lifetime.
```
```suggestion
offlineSessionMaxLimitedHelp=Enable offline session maximum lifetime
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1095,13 +1095,13 @@ clientSaveError=Client could not be updated\: {{error}}
 userGroupsRetrieveStrategyHelp=Specify how to retrieve groups of user. LOAD_GROUPS_BY_MEMBER_ATTRIBUTE means that roles of user will be retrieved by sending LDAP query to retrieve all groups where 'member' is our user. GET_GROUPS_FROM_USER_MEMBEROF_ATTRIBUTE means that groups of user will be retrieved from 'memberOf' attribute of our user or from the other attribute specified by 'Member-Of LDAP A ...
 hour=Hour
-connectionTimeoutHelp=LDAP connection timeout in milliseconds
+connectionTimeoutHelp=LDAP connection timeout in milliseconds.
```
```suggestion
connectionTimeoutHelp=LDAP connection timeout in milliseconds
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1095,13 +1095,13 @@ clientSaveError=Client could not be updated\: {{error}}
+connectionTimeoutHelp=LDAP connection timeout in milliseconds.
 repeat=Repeat
-defaultSigAlgHelp=Default algorithm used to sign tokens for the realm
+defaultSigAlgHelp=Default algorithm used to sign tokens for the realm.
```
```suggestion
defaultSigAlgHelp=Default algorithm used to sign tokens for the realm
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1119,7 +1119,7 @@ deleteRole=Delete this role
 directAccessHelp=This enables support for Direct Access Grants, which means that client has access to username/password of user and exchange it directly with Keycloak server for access token. In terms of OAuth2 specification, this enables support of 'Resource Owner Password Credentials Grant' for this client.
 groupHelp=Group to add the user in. Fill the full path of the group including path. For example\: '/root-group/child-group'.
-clientPolicyNameHelp=Display name of the policy
+clientPolicyNameHelp=Display name of the policy.
```
```suggestion
clientPolicyNameHelp=Display name of the policy
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1162,7 +1162,7 @@ defaultLocale=Default locale
 aggregate.attrs.label=Aggregate attribute values
 removedGroupMembershipError=Error removing group membership: {{error}}
-allowPasswordAuthenticationHelp=Enable/disable possibility of username/password authentication against Kerberos database
+allowPasswordAuthenticationHelp=Enable/disable possibility of username/password authentication against Kerberos database.
```
```suggestion
allowPasswordAuthenticationHelp=Enable/disable possibility of username/password authentication against Kerberos database
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1176,7 +1176,7 @@ eventTypes.CLIENT_INITIATED_ACCOUNT_LINKING_ERROR.name=Client initiated account
 backchannelLogoutSessionRequiredHelp=Specifying whether a sid (session ID) Claim is included in the Logout Token when the Backchannel Logout URL is used.
 global=Global
-userAttributeHelp=Name of user attribute you want to hardcode
+userAttributeHelp=Name of user attribute you want to hardcode.
```
```suggestion
userAttributeHelp=Name of user attribute you want to hardcode
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1194,7 +1194,7 @@ importParseError=Could not parse the file {{error}}
 addMember=Add member
 eventTypes.CLIENT_INFO_ERROR.name=Client info error
-scopeParameterHelp=You can copy/paste this value of scope parameter and use it in initial OpenID Connect Authentication Request sent from this client adapter. Default client scopes and selected optional client scopes will be used when generating token issued for this client
+scopeParameterHelp=You can copy/paste this value of scope parameter and use it in initial OpenID Connect Authentication Request sent from this client adapter. Default client scopes and selected optional client scopes will be used when generating token issued for this client.
```
```suggestion
scopeParameterHelp=You can copy the value of the scope parameter and paste it in the initial OpenID Connect Authentication Request sent from this client adapter. The default client scopes and selected optional client scopes will be used when generating token issued for this client.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1239,13 +1239,13 @@ eventTypes.REVOKE_GRANT_ERROR.description=Revoke grant error
 moveHere=Move here
 noKeys=No keys
-batchSizeHelp=Count of LDAP users to be imported from LDAP to Keycloak within a single transaction
+batchSizeHelp=Count of LDAP users to be imported from LDAP to Keycloak within a single transaction.
```
```suggestion
batchSizeHelp=Count of LDAP users to be imported from LDAP to Keycloak within a single transaction
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1239,13 +1239,13 @@ eventTypes.REVOKE_GRANT_ERROR.description=Revoke grant error
 mapperTypeLdapAttributeMapperHelp=This mapper is supported just if syncRegistrations is enabled. New users registered in Keycloak will be written to the LDAP with the hardcoded value of some specified attribute.
 userRegistrationHelpText=Enable/disable the registration page. A link for registration will show on login page too.
-activeHelp=Set if the keys can be used for signing
+activeHelp=Set if the keys can be used for signing.
```
```suggestion
activeHelp=Set if the keys can be used for signing
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1272,13 +1272,13 @@ userAttribute=User Attribute
 admin=Admin
 syncUsersError=Could not sync users\: '{{error}}'
-generatedAccessTokenHelp=See the example access token, which will be generated and sent to the client when selected user is authenticated. You can see claims and roles that the token will contain based on the effective protocol mappers and role scope mappings and also based on the claims/roles assigned to user himself
+generatedAccessTokenHelp=See the example access token, which will be generated and sent to the client when selected user is authenticated. You can see claims and roles that the token will contain based on the effective protocol mappers and role scope mappings and also based on the claims/roles assigned to user himself.
```
```suggestion
generatedAccessTokenHelp=See the example access token, which will be generated and sent to the client when the selected user is authenticated. You can see claims and roles that the token will contain based on the effective protocol mappers and role scope mappings and also based on the claims and roles assigned to the actual user.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1272,13 +1272,13 @@ userAttribute=User Attribute
+generatedAccessTokenHelp=See the example access token, which will be generated and sent to the client when selected user is authenticated. You can see claims and roles that the token will contain based on the effective protocol mappers and role scope mappings and also based on the claims/roles assigned to user himself.
 webAuthnPolicyAcceptableAaguidsHelp=The list of allowed AAGUIDs of which an authenticator can be registered. An AAGUID is a 128-bit identifier indicating the authenticator's type (e.g., make and model).
-keyPasswordHelp=Password for the private key
+keyPasswordHelp=Password for the private key.
```
```suggestion
keyPasswordHelp=Password for the private key
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1272,13 +1272,13 @@ userAttribute=User Attribute
 clientUpdaterTrustedHostsTooltip=List of Hosts, which are trusted. In case that client registration/update request comes from the host/domain specified in this configuration, condition evaluates to true. You can use hostnames or IP addresses. If you use star at the beginning (for example '*.example.com' ) then whole domain example.com will be trusted.
 titleRoles=Realm roles
-mapperTypeGroupLdapMapperHelp=Used to map group mappings of groups from some LDAP DN to Keycloak group mappings
+mapperTypeGroupLdapMapperHelp=Used to map group mappings of groups from some LDAP DN to Keycloak group mappings.
```
```suggestion
mapperTypeGroupLdapMapperHelp=Used to map group mappings of groups from some LDAP DN to Keycloak group mappings
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1310,12 +1310,12 @@ groupsDescription=A group is a set of attributes and role mappings that can be a
 protocolTypes.all=All
 keyAlias=Key alias
-xContentTypeOptionsHelp=Default value prevents Internet Explorer and Google Chrome from MIME-sniffing a response away from the declared content-type <1>Learn more</1>
+xContentTypeOptionsHelp=Default value prevents Internet Explorer and Google Chrome from MIME-sniffing a response away from the declared content-type. <1>Learn more</1>
```
```suggestion
xContentTypeOptionsHelp=The default value prevents Internet Explorer and Google Chrome from MIME-sniffing a response away from the declared content-type. <1>Learn more</1>
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1326,8 +1326,8 @@ eventTypes.USER_DISABLED_BY_PERMANENT_LOCKOUT.description=User disabled by perma
 eventTypes.USER_DISABLED_BY_TEMPORARY_LOCKOUT_ERROR.description=User disabled by temporary lockout error
-userModelAttributeNameHelp=Name of the model attribute to be added when importing user from LDAP
-templateHelp=Template to use to format the username to import.  Substitutions are enclosed in ${}.  For example\: '${ALIAS}.${CLAIM.sub}'.  ALIAS is the provider alias.  CLAIM.<NAME> references an ID or Access token claim. The substitution can be converted to upper or lower case by appending |uppercase or |lowercase to the substituted value, e.g. '${CLAIM.sub | lowercase}
+userModelAttributeNameHelp=Name of the model attribute to be added when importing user from LDAP.
```
```suggestion
userModelAttributeNameHelp=Name of the model attribute to be added when importing user from LDAP
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1326,8 +1326,8 @@ eventTypes.USER_DISABLED_BY_PERMANENT_LOCKOUT.description=User disabled by perma
-userModelAttributeNameHelp=Name of the model attribute to be added when importing user from LDAP
-templateHelp=Template to use to format the username to import.  Substitutions are enclosed in ${}.  For example\: '${ALIAS}.${CLAIM.sub}'.  ALIAS is the provider alias.  CLAIM.<NAME> references an ID or Access token claim. The substitution can be converted to upper or lower case by appending |uppercase or |lowercase to the substituted value, e.g. '${CLAIM.sub | lowercase}
+userModelAttributeNameHelp=Name of the model attribute to be added when importing user from LDAP.
+templateHelp=Template to use to format the username to import.  Substitutions are enclosed in ${}.  For example\: '${ALIAS}.${CLAIM.sub}'.  ALIAS is the provider alias.  CLAIM.<NAME> references an ID or Access token claim. The substitution can be converted to upper or lower case by appending |uppercase or |lowercase to the substituted value, e.g. '${CLAIM.sub | lowercase}.
```
```suggestion
templateHelp=Template to use to format the username to import.  Substitutions are enclosed in ${}.  For example\: '${ALIAS}.${CLAIM.sub}'.  ALIAS is the provider alias.  CLAIM.<NAME> references an ID or Access token claim. The substitution can be converted to upper or lower case by appending |uppercase or |lowercase to the substituted value, for example, '${CLAIM.sub | lowercase}.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1339,7 +1339,7 @@ passwordDataTitle=Password data
 clientPolicies=Client policies
 NONE=NONE
-keystorePasswordHelp=Password for the keys
+keystorePasswordHelp=Password for the keystore.
```
```suggestion
keystorePasswordHelp=Password for the keystore
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1388,7 +1388,7 @@ authorizationEncryptedResponseAlgHelp=JWA Algorithm used for key management in e
 scopePermissions.users.manage-description=Policies that decide if an administrator can manage all users in the realm
 defaultACRValuesHelp=Default values to be used as voluntary ACR in case that there is no explicit ACR requested by 'claims' or 'acr_values' parameter in the OIDC request.
-minimumACRValueHelp=Minimum ACR to be enforced by keycloak. Overrides lower ACRs explicitly requested via 'acr_values' or 'claims', unless marked they are essential
+minimumACRValueHelp=Minimum ACR to be enforced by keycloak. Overrides lower ACRs explicitly requested via 'acr_values' or 'claims', unless marked they are essential.
```
```suggestion
minimumACRValueHelp=Minimum ACR to be enforced by Keycloak. Overrides lower ACRs explicitly requested by 'acr_values' or 'claims', unless they are marked as essential.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1412,27 +1412,27 @@ noUsersEmptyStateDescriptionContinued=to find them. Users that already have this
 executorsHelpText=Executors, which will be applied for this client profile
 ldapSearchingAndUpdatingSettings=LDAP searching and updating
-authenticationAliasHelp=Name of the configuration
+authenticationAliasHelp=Name of the configuration.
```
```suggestion
authenticationAliasHelp=Name of the configuration
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1412,27 +1412,27 @@ noUsersEmptyStateDescriptionContinued=to find them. Users that already have this
 SSOSessionIdle=SSO Session Idle
 deleteClientPolicyConditionConfirmTitle=Delete condition?
-initialCounterErrorHint=Value needs to be between 1 and 120
+initialCounterErrorHint=Value needs to be between 1 and 120.
```
```suggestion
initialCounterErrorHint=Value needs to be between 1 and 120
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1412,27 +1412,27 @@ noUsersEmptyStateDescriptionContinued=to find them. Users that already have this
 headerName=header name
 issuerHelp=The issuer identifier for the issuer of the response. If not provided, no validation will be performed.
-uiDisplayNameHelp=Display name of provider when linked in the Admin UI
+uiDisplayNameHelp=Display name of provider when linked in the Admin UI.
```
```suggestion
uiDisplayNameHelp=Display name of provider when linked in the Admin UI
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1611,7 +1611,7 @@ attributePermissionDescription=This section contains permissions for who can edi
 groupDeleteError=Error deleting group {{error}}
 editGroupText=Edit attributes group
-updateFirstLoginHelp=Update profile on first login
+updateFirstLoginHelp=Update profile on first login.
```
```suggestion
updateFirstLoginHelp=Update profile on first login
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1641,7 +1641,7 @@ forcePostBinding=Force POST binding
 passMaxAgeHelp=Pass max_age to identity provider.
 exportFail=Could not export realm\: '{{error}}'
-flowTypeHelp=What kind of form is it
+flowTypeHelp=What kind of form is it.
```
```suggestion
flowTypeHelp=What kind of form is it
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1671,7 +1671,7 @@ openIDEndpointConfiguration=OpenID Endpoint Configuration
 prompts.login=Login
 users=Users
-keyTabHelp=Location of Kerberos KeyTab file containing the credentials of server principal. For example, /etc/krb5.keytab
+keyTabHelp=Location of Kerberos KeyTab file containing the credentials of server principal. For example, /etc/krb5.keytab.
```
```suggestion
keyTabHelp=Location of Kerberos KeyTab file containing the credentials of server principal. For example, /etc/krb5.keytab
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1692,7 +1692,7 @@ ldapAttributeValue=LDAP attribute value
 eventTypes.CLIENT_DELETE.description=Client delete
 emptyResources=No resources
-roleHelp=Role to grant to user if all attributes are present. Click 'Select Role' button to browse roles, or just type it in the textbox. To reference a client role the syntax is clientname.clientrole, i.e. myclient.myrole
+roleHelp=Role to grant to user if all attributes are present. Click 'Select Role' button to browse roles, or just type it in the textbox. To reference a client role the syntax is clientname.clientrole, i.e. myclient.myrole.
```
```suggestion
roleHelp=Role to grant to user if all attributes are present. Click 'Select Role' to browse roles, or just type it in the text box. To reference a client role, the syntax is clientname.clientrole, for example, myclient.myrole.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1748,7 +1748,7 @@ replyToDisplayNameHelp=A user-friendly name for the 'Reply-To' address (optional
 signingKeysConfigExplain=If you enable the "Client signature required" below, you must configure the signing keys by generating or importing keys, and the client will sign their saml requests and responses. The signature will be validated.
 newClientProfile=Create client profile
-consoleDisplayConnectionUrlHelp=Connection URL to your LDAP server
+consoleDisplayConnectionUrlHelp=Connection URL to your LDAP server.
```
```suggestion
consoleDisplayConnectionUrlHelp=Connection URL to your LDAP server
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1770,7 +1770,7 @@ useRefreshTokenForClientCredentialsGrant=Use refresh tokens for client credentia
 sectorIdentifierUri.label=Sector Identifier URI
 inputTypeStep=Input step size
-mapperTypeHelp=Used to map single attribute from LDAP user to attribute of UserModel in Keycloak DB
+mapperTypeHelp=Used to map single attribute from LDAP user to attribute of UserModel in Keycloak DB.
```
```suggestion
mapperTypeHelp=Used to map single attribute from LDAP user to attribute of UserModel in Keycloak database
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1793,7 +1793,7 @@ ssoServiceUrl=Single Sign-On service URL
 inputHelperTextAfter=Helper text (under) the input field
 appliedByClients=Applied by the following clients
-createFlowHelp=You can create a top level flow within this from
+createFlowHelp=You can create a top level flow within this from.
```
from or form?

ahus1 replied:
It is `form`, I'll correct it in the next commit.

```suggestion
createFlowHelp=You can create a top level flow within this form.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1858,7 +1858,7 @@ requestObject.not\ required=Not required
 generatedAccessTokenNo=No generated access token
 always=Always
-pkceEnabledHelp=Use PKCE (Proof of Key-code exchange) for IdP Brokering
+pkceEnabledHelp=Use PKCE (Proof of Key-code exchange) for IdP Brokering.
```
```suggestion
pkceEnabledHelp=Use PKCE (Proof of Key-code exchange) for IdP Brokering
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1954,10 +1954,10 @@ reqAuthnConstraints=Requested AuthnContext Constraints
 addIdpMapperNameHelp=Name of the mapper.
 requirements.ALTERNATIVE=Alternative
-claimFilterValueHelp=Value of the essential claim (with regex support)
+claimFilterValueHelp=Value of the essential claim (with regex support).
```
```suggestion
claimFilterValueHelp=Value of the essential claim (with regex support)
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -1983,7 +1983,7 @@ createClientProfileError=Could not create client profile\: '{{error}}'
 targetOptions.local=LOCAL
 addTranslationError=Error creating translation, {{error}}
-pkceMethodHelp=PKCE Method to use
+pkceMethodHelp=PKCE Method to use.
```
```suggestion
pkceMethodHelp=PKCE Method to use
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2001,7 +2001,7 @@ selectMethod=Select method
 eventTypes.VALIDATE_ACCESS_TOKEN_ERROR.name=Validate access token error
 xFrameOptions=X-Frame-Options
-scopeDescriptionHelp=Description of the client scope
+scopeDescriptionHelp=Description of the client scope.
```
```suggestion
scopeDescriptionHelp=Description of the client scope
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2276,7 +2276,7 @@ eventTypes.OAUTH2_DEVICE_VERIFY_USER_CODE.description=OAuth2 device verify user
 validateUuidLDAPAttribute=You must enter a UUID LDAP attribute
 client-scopes-condition.tooltip=The list of expected client scopes. Condition evaluates to true if specified client request matches some of the client scopes. It depends also whether it should be default or optional client scope based on the 'Scope Type' configured.
-rootURLHelp=Root URL appended to relative URLs
+rootURLHelp=Root URL appended to relative URLs.
```
```suggestion
rootURLHelp=Root URL appended to relative URLs
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2303,7 +2303,7 @@ requiredUserActionsHelp=Require an action when the user logs in. 'Verify email'
 ldapKerberosSettingsDescription=This section contains options useful for the Kerberos integration. This is used only when the LDAP server is used together with Kerberos/SPNEGO for user authentication.
 deleteEvents=Clear events
-termsOfServiceUrlHelp=URL that the Relying Party Client provides to the End-User to read about the Relying Party's terms of service
+termsOfServiceUrlHelp=URL that the Relying Party Client provides to the End-User to read about the Relying Party's terms of service.
```
```suggestion
termsOfServiceUrlHelp=URL that the Relying Party Client provides to the End-User to read about the Relying Party's terms of service
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2312,7 +2312,7 @@ downloadAdaptorTitle=Download adaptor configs
 keysFilter.PASSIVE=Passive keys
 revocation=Revocation
-scopeTypeHelp=Client scopes, which will be added as default scopes to each created client
+scopeTypeHelp=Client scopes, which will be added as default scopes to each created client.
```
```suggestion
scopeTypeHelp=Client scopes, which will be added as default scopes to each created client
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2323,13 +2323,13 @@ permissionDeletedError=Could not delete permission due to {{error}}
 realmId=Realm ID
 eventTypes.PERMISSION_TOKEN_ERROR.description=Permission token error
-algorithmHelp=Intended algorithm for the key
+algorithmHelp=Intended algorithm for the key.
```
```suggestion
algorithmHelp=Intended algorithm for the key
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2323,13 +2323,13 @@ permissionDeletedError=Could not delete permission due to {{error}}
 displayOnClientHelp=Applicable only if 'Consent Required' is on for this client. If this switch is off, the consent screen will contain just the consents corresponding to configured client scopes. If on, there will be also one item on the consent screen about this client itself.
 requestObjectRequired=Request object required
-protocolHelp=Which SSO protocol configuration is being supplied by this client scope
+protocolHelp=Which SSO protocol configuration is being supplied by this client scope.
```
```suggestion
protocolHelp=Which SSO protocol configuration is being supplied by this client scope
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2356,15 +2356,15 @@ frontendUrlHelp=Set the frontend URL for the realm. Use in combination with the
 eventListeners=Event listeners
 bindDn=Bind DN
-evictionHourHelp=Hour of the day the entry will become invalid
+evictionHourHelp=Hour of the day the entry will become invalid.
```
```suggestion
evictionHourHelp=Hour of the day the entry will become invalid
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2403,7 +2403,7 @@ roleNameLdapAttributeHelp=Name of LDAP attribute, which is used in role objects
 regexPattern=Regex pattern
 targetContextAttributes=Target Context Attributes
-targetContextAttributesHelp=Defines the evaluation of context attributes (claims) instead of identity attributes
+targetContextAttributesHelp=Defines the evaluation of context attributes (claims) instead of identity attributes.
```
```suggestion
targetContextAttributesHelp=Defines the evaluation of context attributes (claims) instead of identity attributes
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2423,12 +2423,12 @@ addSubFlowHelp=Sub-Flows can be either generic or form. The form type is used to
 authorizationSignedResponseAlgHelp=JWA algorithm used for signing authorization response tokens when the response mode is jwt.
 associatedRolesRemoved=Associated roles have been removed
-keyAliasHelp=Alias for the private key
+keyAliasHelp=Alias for the private key.
```
```suggestion
keyAliasHelp=Alias for the private key
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2423,12 +2423,12 @@ addSubFlowHelp=Sub-Flows can be either generic or form. The form type is used to
 userName=Username
 clientProfileDescription=Description
-ellipticCurveHelp=Elliptic curve used in ECDSA
+ellipticCurveHelp=Elliptic curve used in ECDSA.
```
```suggestion
ellipticCurveHelp=Elliptic curve used in ECDSA
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2476,7 +2476,7 @@ saveEventListenersError=Error saving event listener\: {{error}}
 multivalued.tooltip=Indicates if attribute supports multiple values. If true, the list of all values of this attribute will be set as claim. If false, just first value will be set as claim
 inputOptionLabelsI18nPrefix=Internationalization key prefix
-enabledHelp=Set if the keys are enabled
+enabledHelp=Set if the keys are enabled.
```
```suggestion
enabledHelp=Set if the keys are enabled
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2531,7 +2531,7 @@ eventTypes.IDENTITY_PROVIDER_POST_LOGIN_ERROR.description=Identity provider post
 mappingDeletedSuccess=Mapping successfully deleted
 addIdentityProvider=Add {{provider}} provider
-flowDescriptionHelp=Help text for the description of the new flow
+flowDescriptionHelp=Help text for the description of the new flow.
```
```suggestion
flowDescriptionHelp=Help text for the description of the new flow
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2590,7 +2590,7 @@ policyEnforcementModes.DISABLED=Disabled
 no=No
 code=Code
-nameHelp=Help text for the name of the new flow
+nameHelp=Help text for the name of the new flow.
```
```suggestion
nameHelp=Help text for the name of the new flow
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2601,14 +2601,14 @@ isBinaryAttributeHelp=Should be true for binary LDAP attributes.
 eventTypes.RESTART_AUTHENTICATION_ERROR.name=Restart authentication error
 generatedUserInfoIsDisabled=Generated user info is disabled when no user is selected
-nameHelpHelp=Name of the mapper
+nameHelpHelp=Name of the mapper.
```
```suggestion
nameHelpHelp=Name of the mapper
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2601,14 +2601,14 @@ isBinaryAttributeHelp=Should be true for binary LDAP attributes.
 createClientConditionError=Error creating condition\: {{error}}
-serverPrincipalHelp=Full name of server principal for HTTP service including server and domain name. For example, HTTP/host.foo.org@FOO.ORG
-enableStartTlsHelp=Encrypts the connection to LDAP using STARTTLS, which will disable connection pooling
+serverPrincipalHelp=Full name of server principal for HTTP service including server and domain name. For example, HTTP/host.foo.org@FOO.ORG.
```
```suggestion
serverPrincipalHelp=Full name of server principal for HTTP service including server and domain name. For example, HTTP/host.foo.org@FOO.ORG
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2601,14 +2601,14 @@ isBinaryAttributeHelp=Should be true for binary LDAP attributes.
-serverPrincipalHelp=Full name of server principal for HTTP service including server and domain name. For example, HTTP/host.foo.org@FOO.ORG
-enableStartTlsHelp=Encrypts the connection to LDAP using STARTTLS, which will disable connection pooling
+serverPrincipalHelp=Full name of server principal for HTTP service including server and domain name. For example, HTTP/host.foo.org@FOO.ORG.
+enableStartTlsHelp=Encrypts the connection to LDAP using STARTTLS, which will disable connection pooling.
```
```suggestion
enableStartTlsHelp=Encrypts the connection to LDAP using STARTTLS, which will disable connection pooling
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2669,12 +2669,12 @@ eventTypes.USER_INFO_REQUEST.description=User info request
 resultDeny=Result-Deny
 kc.client.network.host=Client Host
-noResourceCreateHint=There are no resources you can't create resource-based permission
+noResourceCreateHint=There are no resources you can't create resource-based permission.
```
```suggestion
noResourceCreateHint=There are no resources. Therefore, you cannot create resource-based permissions.
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2755,7 +2755,7 @@ passwordPolicyHintsEnabled=Password policy hints enabled
 syncMode=Sync mode
 details=Details
-privateRSAKeyHelp=Private RSA Key encoded in PEM format
+privateRSAKeyHelp=Private RSA Key encoded in PEM format.
```
```suggestion
privateRSAKeyHelp=Private RSA Key encoded in PEM format
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2847,7 +2847,7 @@ orderDialogIntro=The order that the providers are listed in the login page or th
 push=Push
 targetClaimHelp=Specifies the target claim which the policy will fetch.
-periodicFullSyncHelp=Whether periodic full synchronization of LDAP users to Keycloak should be enabled or not
+periodicFullSyncHelp=Whether periodic full synchronization of LDAP users to Keycloak should be enabled or not.
```
```suggestion
periodicFullSyncHelp=Whether periodic full synchronization of LDAP users to Keycloak should be enabled or not
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -3182,7 +3182,7 @@ parRequestUriLifespan=Lifetime of the Request URI for Pushed Authorization Reque
 identityBrokeringLink=Identity brokering link
 searchClientRegistration=Search for policy
-importFileHelp=File to import a key
+importFileHelp=File to import a key.
```
```suggestion
importFileHelp=File to import a key
```

andymunro commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -3252,7 +3252,7 @@ linkedOrganization=Linked organization
 send=Send
 redirectWhenEmailMatches=Redirect when email domain matches
-redirectWhenEmailMatchesHelp=Automatically redirect the user to this identity provider when the email domain matches the domain
+redirectWhenEmailMatchesHelp=Automatically redirect the user to this identity provider when the email domain matches the domain.
```
```suggestion
redirectWhenEmailMatchesHelp=Automatically redirect the user to this identity provider when the email domain matches the domain
```

andymunro reviewed (changes requested):
Looks good.  I have a few suggestions, @ahus1.  I am happy to discuss where we do not have periods for sentence fragments if perhaps they are mixed in with messages in the same area and look incomplete.

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -188,7 +188,7 @@ eventTypes.LOGOUT.description=Logout
 lifespan=Expires In
 storedTokensReadableHelp=Enable/disable if new users can read any stored tokens. This assigns the broker.read-token role.
-authenticationFlowTypeHelp=What kind of form is it
+authenticationFlowTypeHelp=What kind of form is it.
```
```suggestion
authenticationFlowTypeHelp=What kind of form is it?
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -109,7 +109,7 @@ updateExecutorSuccess=Executor updated successfully
 assertionLifespan=Assertion Lifespan
 export=Export
-claimFilterNameHelp=Name of the essential claim
+claimFilterNameHelp=Name of the essential claim.
```
```suggestion
claimFilterNameHelp=Name of the essential claim
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -241,12 +241,12 @@ generatedAccessTokenIsDisabled=Generated access token is disabled when no user i
 userInfoResponseEncryptionKeyManagementAlgorithm=User info response encryption key management algorithm
 changedUsersSyncPeriod=Changed users sync period
-keystoreHelp=Path to keys file
+keystoreHelp=Path to keys file.
```
```suggestion
keystoreHelp=Path to keys file
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -241,12 +241,12 @@ generatedAccessTokenIsDisabled=Generated access token is disabled when no user i
 helpFileUploadClient=Upload a JSON or XML file
 generateSuccess=New key pair and certificate generated successfully
-userAttributeValueHelp=Value you want to hardcode
+userAttributeValueHelp=Value you want to hardcode.
```
```suggestion
userAttributeValueHelp=Value you want to hardcode
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -339,7 +339,7 @@ backchannelLogoutSessionRequired=Backchannel logout session required
 eventTypes.PUSHED_AUTHORIZATION_REQUEST_ERROR.name=Pushed authorization request error
 editAttribute=Edit attribute
-webAuthnPolicyRpEntityNameHelp=Human-readable server name as WebAuthn Relying Party
+webAuthnPolicyRpEntityNameHelp=Human-readable server name as WebAuthn Relying Party.
```
```suggestion
webAuthnPolicyRpEntityNameHelp=Human-readable server name as WebAuthn Relying Party
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -415,7 +415,7 @@ joinGroup=Join Group
 eventTypes.EXECUTE_ACTION_TOKEN_ERROR.description=Execute action token error
 unlinkAccountConfirm=Are you sure you want to permanently unlink this account from {{provider}}?
-x509CertificateHelp=X509 Certificate encoded in PEM format
+x509CertificateHelp=X509 Certificate encoded in PEM format.
```
```suggestion
x509CertificateHelp=X509 Certificate encoded in PEM format
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -442,10 +442,10 @@ refresh=Refresh
 advancedClaimToRole=If all claims exist, grant the user the specified realm or client role.
 directGrant=Direct Grant Flow
-maxLifespanHelp=Max lifespan of cache entry in milliseconds
+maxLifespanHelp=Max lifespan of cache entry in milliseconds.
```
```suggestion
maxLifespanHelp=Max lifespan of cache entry in milliseconds
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -465,7 +465,7 @@ evictionHour=Eviction hour
 otpSupportedApplications.totpAppMicrosoftAuthenticatorName=Microsoft Authenticator
 ldapMappersList=LDAP Mappers
-bindDnHelp=DN of the LDAP admin, which will be used by Keycloak to access LDAP server
+bindDnHelp=DN of the LDAP admin, which will be used by Keycloak to access LDAP server.
```
```suggestion
bindDnHelp=DN of the LDAP admin, which will be used by Keycloak to access LDAP server
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -487,7 +487,7 @@ eventTypes.CLIENT_DELETE_ERROR.description=Client delete error
 revokeRefreshToken=Revoke Refresh Token
 mappingUpdatedSuccess=Mapping successfully updated
-logoUrlHelp=URL that references a logo for the Client application
+logoUrlHelp=URL that references a logo for the Client application.
```
```suggestion
logoUrlHelp=URL that references a logo for the Client application
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -542,9 +542,9 @@ otpPolicyCodeReusable=Reusable token
+mapperTypeRoleLdapMapperHelp=Used to map role mappings of roles from some LDAP DN to Keycloak role mappings of either realm roles or client roles of particular client.
 used.DEFAULT=Default
-authenticationCreateFlowHelp=Create flow
+authenticationCreateFlowHelp=Create flow.
```
```suggestion
authenticationCreateFlowHelp=Create flow
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -567,10 +567,10 @@ eventTypes.EXECUTE_ACTION_TOKEN.description=Execute action token
 sessions=Sessions
 mapperCreateSuccess=Mapper created successfully.
-fullSyncPeriodHelp=Period for full synchronization in seconds
+fullSyncPeriodHelp=Period for full synchronization in seconds.
```
```suggestion
fullSyncPeriodHelp=Period for full synchronization in seconds
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -651,10 +651,10 @@ deleteResource=Permanently delete resource?
 emptyAddClientScopesInstructions=There are no client scopes left to add
 changeTypeIntro={{count}} selected client scopes will be changed to
-secretSizeHelp=Size in bytes for the generated secret
+secretSizeHelp=Size in bytes for the generated secret.
```
```suggestion
secretSizeHelp=Size in bytes for the generated secret
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -694,7 +694,7 @@ webAuthnPolicyCreateTimeoutHelp=The timeout value for creating the user's public
 claimFilterValue=Essential claim value
 eventTypes.REGISTER_ERROR.name=Register error
-priorityHelp=Priority of the provider
+priorityHelp=Priority of the provider.
```
```suggestion
priorityHelp=Priority of the provider
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -758,17 +758,17 @@ otpTypeHelp=totp is Time-Based One Time Password. 'hotp' is a counter base one t
 path=Path
 overwritten=Overwritten
-mapperNameHelp=Name of the mapper
+mapperNameHelp=Name of the mapper.
```
```suggestion
mapperNameHelp=Name of the mapper
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -758,17 +758,17 @@ otpTypeHelp=totp is Time-Based One Time Password. 'hotp' is a counter base one t
+mapperNameHelp=Name of the mapper.
 keyProviderMapperNameHelp=Display name of provider when linked in admin console.
-providerIdHelp=Provider ID
+providerIdHelp=Provider ID.
```
```suggestion
providerIdHelp=Provider ID
```

ahus1 commented on js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:
```diff
@@ -2356,7 +2356,7 @@ frontendUrlHelp=Set the frontend URL for the realm. Use in combination with the
 emailSettings=Email settings
 samlEntityDescriptorHelp=Allows you to load external IDP metadata from a config file or to download it from a URL.
-generatedIdTokenHelp=See the example ID Token, which will be generated and sent to the client when selected user is authenticated. You can see claims and roles that the token will contain based on the effective protocol mappers and role scope mappings and also based on the claims/roles assigned to user himself
+generatedIdTokenHelp=See the example ID Token, which will be generated and sent to the client when the elected user is authenticated. You can see claims and roles that the token will contain based on the effective protocol mappers and role scope mappings and also based on the claims and roles assigned to the actual user.
```
```suggestion
generatedIdTokenHelp=See the example ID Token, which will be generated and sent to the client when the selected user is authenticated. You can see claims and roles that the token will contain based on the effective protocol mappers and role scope mappings and also based on the claims and roles assigned to the actual user.
```

ahus1 commented:
Thanks to @andymunro, those incomplete sentences don't have a full stop at the end any more. He also fixed several other wording and naming issues, thank you for that!

So I'd argue this is now ready for review, and with those fixes it is worth merging. 

@ssilvert - please give it another round of review. 

Thanks!
